_posts/2020-06-05-cruise-control.md

```diff
@@ -0,0 +1,213 @@
+
+- `goals`: This configures the _master_ goals list. Basically, this is the list of goals that are allowed to be used in any of Cruise Control's functions. You would use this primarily to limit the optimization goals that that users of your cluster's Cruise Control server can use. For example if, for some reason, you don't want any cluster user to use the preferred leader election goal, you can define a `goals` list that does not include it.
+- `hard.goals`: This list of goals will define which goals in the master goals list are considered _hard_ and cannot be violated in any of the optimization functions of Cruise Control. The longer this list, the less likely it is that Cruise Control will be able to find a viable optimization proposal, so if you do customise this, try to make it as short as possible.
+- `default.goals`: The default goals list is the most important of these configurations. By default, every 15 mins, Cruise Control will use the current state of your Kafka cluster to generate a _cached optimization proposal_ using the configured `default.goals` list. This means you should configure the `default.goals` list to be the set of goals that you will most often want your cluster to meet. That way, when you ask for a rebalance, a optimization proposal should be already prepared.
+
+All these configuration options can be set under `Kafka.spec.cruiseControl.config`. Please consult the [Strimzi documentation](https://strimzi.io/docs/operators/latest/using.html#con-optimization-goals-str) for more detailed information on the various goal configurations.
+
+### Getting an optimization proposal
+
+Now you have Cruise Control deployed you can ask it to generate a optimization proposal. To do this you will need to apply a `KafkaRebalance` custom resource (the definition for this is installed when you install Strimzi). A basic `KafkaRebalance` is shown below:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+  spec: {}
+```
+
+And you can deploy this like any other resource:
+
+```bash
+kubectl apply -f kafka-rebalance.yaml -n myproject
+```
+
+This rebalance has an empty `spec` field and so will use the default rebalance settings. This means it will use the `default.goals` (mentioned above) and will return the most recent _cached optimization proposal_. Once you apply this resource to your Kafka cluster the Cluster Operator will issue the relevant requests to Cruise Control to fetch the optimization proposal. The Cluster Operator will keep the status of your `KafkaRebalance` updated with the progress from Cruise Control. You can check this using:
+
+```bash
+$ kubectl describe kafkarebalance -n myproject my-rebalance      
```

Comment:
I would put the namespace parameter at the end. I think having `kafkarebalance my-rebalance` is more readable.